ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/netty/NettyTLSConnectionInitializerTest.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.p2p.rlpx.connections.netty;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.crypto.NodeKey;
+import org.hyperledger.besu.ethereum.p2p.config.RlpxConfiguration;
+import org.hyperledger.besu.ethereum.p2p.peers.LocalNode;
+import org.hyperledger.besu.ethereum.p2p.rlpx.connections.PeerConnectionEventDispatcher;
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+import io.netty.handler.codec.LengthFieldPrepender;
+import io.netty.handler.codec.compression.SnappyFrameDecoder;
+import io.netty.handler.codec.compression.SnappyFrameEncoder;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslHandler;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NettyTLSConnectionInitializerTest {
+
+  @Mock private NodeKey nodeKey;
+  @Mock private RlpxConfiguration rlpxConfiguration;
+  @Mock private LocalNode localNode;
+  @Mock private PeerConnectionEventDispatcher eventDispatcher;
+  @Mock private TLSContextFactory tlsContextFactory;
+  @Mock private SslContext sslContext;
+  @Mock private SslHandler sslHandler;
+
+  private NettyTLSConnectionInitializer nettyTLSConnectionInitializer;
+
+  @Before
+  public void before() throws Exception {
+    nettyTLSConnectionInitializer =
+        new NettyTLSConnectionInitializer(
+            nodeKey,
+            rlpxConfiguration,
+            localNode,
+            eventDispatcher,
+            new NoOpMetricsSystem(),
+            () -> tlsContextFactory);
+
+    when(tlsContextFactory.createNettyServerSslContext()).thenReturn(sslContext);
+    when(tlsContextFactory.createNettyClientSslContext()).thenReturn(sslContext);
+    when(sslContext.newHandler(any())).thenReturn(sslHandler);
+  }
+
+  @Test
+  public void addAdditionalOutboundHandlersIncludesAllExpectedHandlersToChannelPipeline()
+      throws Exception {
+    final EmbeddedChannel embeddedChannel = new EmbeddedChannel();
+    nettyTLSConnectionInitializer.addAdditionalOutboundHandlers(embeddedChannel);
+
+    // TLS
+    assertThat(embeddedChannel.pipeline().get(SslHandler.class)).isNotNull();
+
+    // Snappy compression
+    assertThat(embeddedChannel.pipeline().get(SnappyFrameDecoder.class)).isNotNull();
+    assertThat(embeddedChannel.pipeline().get(SnappyFrameEncoder.class)).isNotNull();
+
+    // Message Framing
+    assertThat(embeddedChannel.pipeline().get(LengthFieldBasedFrameDecoder.class)).isNotNull();
+    assertThat(embeddedChannel.pipeline().get(LengthFieldPrepender.class)).isNotNull();
```

Comment:
This is the only blocker for me

Reply:
Instead of replacing the current tests and checking only the order based on names, I will add the order check as part of the test. This way we can guarantee that the actual type of the handlers are correct, and that they are in the expected order.